pkg/engine/armtype.go

```diff
@@ -60,6 +63,40 @@ type AvailabilitySetARM struct {
 	compute.AvailabilitySet
 }
 
+// MarshalJSON is the custom marshaler for an AvailabilitySetARM.
+// It acts as a decorator by replacing the JSON field "platformFaultDomainCount"
```

Comment:
Alternatives considered here:

- `json.Marshal` an anonymous struct that matches `AvailabilitySetARM` except that it has a string in the `platformFaultDomainField`. This is a common approach to mutating default output in a `MarshalJSON` function, but it's verbose and would break if a `compute.AvailabilitySet` ever changes.
- Decorate the `compute.AvailabilitySet` struct instead, or perhaps its `Properties` embedded struct where the field we care about actually resides. The default behavior for those structs is in vendored classes, and while it may be possible to type alias one of those structs (as we do below) to avoid blowing the stack with infinite recursion, I couldn't figure out how.
- Just change our `AvailabilitySetARM` to use a wrapper class around `compute.AvailabilitySet` that exposes `platformFaultDomainCount` as a string rather than an int32. Then have the default be our funky ARM expression and an actual value be a simple expression like `"[toString(3)]"` and set the value on the availability set in the apimodel before template generation. Simple...except availability sets aren't really in the model, they're really just template entities.